FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/src/tls_freertos.c

```diff
@@ -510,32 +500,49 @@ TlsTransportStatus_t TLS_FreeRTOS_Connect( NetworkContext_t * pNetworkContext,
 
 void TLS_FreeRTOS_Disconnect( NetworkContext_t * pNetworkContext )
 {
-    BaseType_t socketStatus = 0;
+    BaseType_t tlsStatus = 0;
 
-    socketStatus = ( BaseType_t ) mbedtls_ssl_close_notify( &( pNetworkContext->sslContext.context ) );
+    /* Attempting to terminate TLS connection. */
+    tlsStatus = ( BaseType_t ) mbedtls_ssl_close_notify( &( pNetworkContext->sslContext.context ) );
 
     /* Ignore the WANT_READ and WANT_WRITE return values. */
-    if( ( socketStatus != ( BaseType_t ) MBEDTLS_ERR_SSL_WANT_READ ) &&
-        ( socketStatus != ( BaseType_t ) MBEDTLS_ERR_SSL_WANT_WRITE ) )
+    if( ( tlsStatus != ( BaseType_t ) MBEDTLS_ERR_SSL_WANT_READ ) &&
+        ( tlsStatus != ( BaseType_t ) MBEDTLS_ERR_SSL_WANT_WRITE ) )
     {
-        if( socketStatus == 0 )
+        if( tlsStatus == 0 )
         {
             LogInfo( ( "(Network connection %p) TLS close-notify sent.",
                        pNetworkContext ) );
         }
         else
         {
-            LogError( ( "(Network connection %p) Failed to send TLS close-notify, error %d.",
+            LogError( ( "(Network connection %p) Failed to send TLS close-notify: mbedTLSError= %s : %s.",
                         pNetworkContext,
-                        socketStatus ) );
+                        mbedtlsHighLevelCodeOrDefault( tlsStatus ),
+                        mbedtlsLowLevelCodeOrDefault( tlsStatus ) ) );
         }
     }
+    else
+    {
+        /* WANT_READ and WANT_WRITE can be ignored. Logging for debugging purposes. */
+        LogInfo( ( "(Network connection %p) TLS close-notify sent; ",
```

Comment:
Should this be logged at warning level?
```suggestion
        LogWarn( ( "(Network connection %p) TLS close-notify sent; ",
```

Reply:
I thought about it again and since it is a happy path, info would just be good is what I think.